internal/models.py

```diff
@@ -0,0 +1,174 @@
+from django.contrib.auth.models import User, Group
+from django.db import models
+from django.db.models.signals import post_save, m2m_changed
+from django.dispatch import receiver
+from django.urls import reverse
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from groups.models import Committee
+from internal.util import date_to_term
+
+
+class Member(models.Model):
+    class Meta:
+        permissions = (
+            ("is_internal", "Is a member of MAKE NTNU"),
+            ("can_register_new_member", "Can register new member"),
+            ("can_edit_group_membership", "Can edit the groups a member is part of, including (de)activation")
+        )
+
+    user = models.OneToOneField(User, on_delete=models.DO_NOTHING, null=True, verbose_name=_("User"))
+    committees = models.ManyToManyField(Committee, blank=True, verbose_name=_("Committees"))
+    role = models.CharField(max_length=64, blank=True, verbose_name=_("Role"))
+    email = models.EmailField(blank=True, null=True, verbose_name=_("Email"))
+    phone_number = models.CharField(max_length=32, default="", blank=True, verbose_name=_("Phone number"))
+    study_program = models.CharField(max_length=32, default="", blank=True, verbose_name=_("Study program"))
+    card_number = models.CharField(max_length=32, default="", blank=True, verbose_name=_("Card number (EM)"))
+    date_joined = models.DateField(default=timezone.datetime.now, verbose_name=_("Date joined"))
+    date_quit = models.DateField(blank=True, null=True, verbose_name=_("Date quit"))
+    reason_quit = models.TextField(max_length=256, default="", blank=True, verbose_name=_("Reason quit"))
+    comment = models.TextField(max_length=256, default="", blank=True, verbose_name=_("Comment"))
+    active = models.BooleanField(default=True, verbose_name=_("Is active"))
+    guidance_exemption = models.BooleanField(default=False, verbose_name=_("Guidance exemption"))
+    quit = models.BooleanField(default=False, verbose_name=_("Has quit"))
+    retired = models.BooleanField(default=False, verbose_name=_("Retired"))
+    honorary = models.BooleanField(default=False, verbose_name=_("Honorary"))
+
+    def on_create(self):
```

Comment:
I find it slightly messy and hard to read when this is called by an external receiver. You can do the same in save() using `if self.pk == None`

Reply:
Yeah, the receiver logic for creation is remnant from when updates to the committées went through it. I agree that an override in `save()` is better here now.